src/org/zaproxy/zap/extension/websocket/WebSocketSenderScriptListener.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * 
+ */
+package org.zaproxy.zap.extension.websocket;
+
+import java.util.List;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.zaproxy.zap.extension.script.ExtensionScript;
+import org.zaproxy.zap.extension.script.ScriptWrapper;
+import org.zaproxy.zap.extension.websocket.WebSocketProxy.Initiator;
+import org.zaproxy.zap.extension.websocket.WebSocketProxy.State;
+
+/**
+ * @author Juha Kivekas
+ *
+ */
+public class WebSocketSenderScriptListener implements WebSocketSenderListener {
+	
+	private ExtensionScript extensionScript;
+
+	public WebSocketSenderScriptListener(){
+		return;
+	}
+
+	private ExtensionScript getExtension() {
+		if (extensionScript == null) {
+			extensionScript = Control.getSingleton().getExtensionLoader().getExtension(ExtensionScript.class);
+		}
+		return extensionScript;
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.zaproxy.zap.extension.websocket.WebSocketSenderListener#getListenerOrder()
+	 */
+	@Override
+	public int getListenerOrder() {
+		// TODO Auto-generated method stub
+		return 0;
+	}
+
+	/* (non-Javadoc)
+	 * @see org.zaproxy.zap.extension.websocket.WebSocketSenderListener#onMessageFrame(int, org.zaproxy.zap.extension.websocket.WebSocketMessage, org.zaproxy.zap.extension.websocket.WebSocketProxy.Initiator)
+	 */
+	@Override
+	public void onMessageFrame(int channelId, WebSocketMessage message,
+			Initiator initiator) {
+		if (this.getExtension() != null) {
```

Comment:
This could be removed, the listener is created/added only if the extension is available.